rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/LocalInfrastructure.java

```diff
@@ -120,51 +107,68 @@ private void acquireNodeImpl() {
             Collections.addAll(paPropList, this.paProperties.split(" "));
         }
         clb.setPaProperties(paPropList);
-        clb.setNodeName(nodeName);
+        clb.setNodeName(baseNodeName);
+        clb.setNumberOfNodes(handledNodes.get());
         try {
             clb.setCredentialsValueAndNullOthers(new String(this.credentials.getBase64()));
         } catch (KeyException e) {
-            createLostNode(nodeName, "Cannot decrypt credentials value", e);
+            createLostNodes(baseNodeName, "Cannot decrypt credentials value", e);
             return;
         }
         List<String> cmd;
         try {
             cmd = clb.buildCommandLineAsList(false);
         } catch (IOException e) {
-            createLostNode(nodeName, "Cannot build command line", e);
+            createLostNodes(baseNodeName, "Cannot build command line", e);
             return;
         }
 
         // The printed cmd with obfuscated credentials
         final String obfuscatedCmd = Tools.join(cmd, " ");
 
-        String depNodeURL = null;
-        ProcessExecutor processExecutor;
+        List<String> depNodeURLs = new ArrayList<>(handledNodes.get());
         try {
-            this.isNodeAcquired.put(nodeName, false);
-            depNodeURL = this.addDeployingNode(nodeName, obfuscatedCmd, "Node launched locally",
-                    this.nodeTimeout);
+            List<String> createdNodeNames = RMNodeStarter.getWorkersNodeNames(baseNodeName, handledNodes.get());
+            for (int nodeIndex = 0; nodeIndex < handledNodes.get(); nodeIndex++) {
+                String depNodeURL = this.addDeployingNode(createdNodeNames.get(nodeIndex), obfuscatedCmd, "Node launched locally",
+                        this.nodeTimeout);
+                depNodeURLs.add(depNodeURL);
+            }
 
             // Deobfuscate the cred value
             Collections.replaceAll(cmd, CommandLineBuilder.OBFUSC, clb.getCredentialsValue());
 
-            this.isDeployingNodeLost.put(depNodeURL, false);
-            processExecutor = new ProcessExecutor(nodeName, cmd, false, true);
+            processExecutor = new ProcessExecutor(baseNodeName, cmd, false, true);
             processExecutor.start();
 
-            this.nodeNameToProcess.put(nodeName, processExecutor);
+            Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    if (processExecutor != null) {
+                        if (!processExecutor.isProcessFinished()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)